frontend-web/src/app/components/ui/AccountWidget.tsx

```diff
@@ -0,0 +1,21 @@
+'use client';
+
+import React, { useEffect, useState } from 'react';
+
+import { User, withAuth } from '@/utils/withAuth';
+
+import LoginButton from './LoginButton';
+import UserMenu from './UserMenu';
+
+export default function AccountWidget(): React.ReactElement {
+  const [user, setUser] = useState<User | null>(null);
```

Comment:
tu peux retirer le useEffect et le useState

et récupérer l'objet user si tu modifies frontend-web/src/utils/withAuth.ts

```suggestion
  let user: User | undefined = undefined;
  const { user } = useWithAuth();
  
  return user !== undefined ? <UserMenu user={user} /> : <LoginButton />;
```